lib/mappers/figma/gradient.js

```diff
@@ -0,0 +1,40 @@
+const formatColor = require('../sketch/color');
+const { tokenize, mapColorKeys } = require('../../utils');
+
+// Property locations differ with regular gradients and shadow gradients
+const getGradientPropertyLocation = gradient =>
+  gradient.gradientHandlePositions ? gradient : gradient.children[0].fills[0];
+
+const getGradientStops = gradient => {
+  const { gradientStops } = getGradientPropertyLocation(gradient);
+  return {
+    stops:
+      gradientStops !== undefined
+        ? gradientStops.map(stop => ({
+            position: stop.position,
+            color: formatColor(mapColorKeys(stop.color)),
+          }))
+        : null,
+  };
+};
+
+const getGradientLocations = gradient => {
+  const { gradientHandlePositions } = getGradientPropertyLocation(gradient);
+  const lastGradient = gradientHandlePositions
+    ? gradientHandlePositions[gradientHandlePositions.length - 1]
+    : undefined;
+  return {
+    from: gradientHandlePositions
+      ? [gradientHandlePositions[0].x, gradientHandlePositions[0].y]
+      : null,
+    to: lastGradient ? [lastGradient.x, lastGradient.y] : null,
+  };
+};
+
+module.exports = gradient => ({
+  name: gradient.name ? gradient.name.split('/').pop() : null,
```

Comment:
names should never be null, if we have no name this is not a valid gradient and the program should crash

Reply:
I did this because a border gradient in Figma doesn't have a name. In Sketch this border gradient is called `norwayGradient`.

```json
{
            "id": "1:60",
            "name": "primitives/border/gradient-with-corners",
            "type": "FRAME",
            "blendMode": "PASS_THROUGH",
            "children": [
              {
                "id": "1:61",
                "name": "Rectangle",
                "type": "RECTANGLE",
                "blendMode": "PASS_THROUGH",
                "absoluteBoundingBox": {
                  "x": 336,
                  "y": 54,
                  "width": 80,
                  "height": 25
                },
                "constraints": {
                  "vertical": "SCALE",
                  "horizontal": "SCALE"
                },
                "fills": [
                  {
                    "blendMode": "NORMAL",
                    "type": "SOLID",
                    "color": {
                      "r": 0.847000002861023,
                      "g": 0.847000002861023,
                      "b": 0.847000002861023,
                      "a": 1
                    }
                  }
                ],
                "strokes": [
                  {
                    "blendMode": "NORMAL",
                    "type": "GRADIENT_LINEAR",
                    "gradientHandlePositions": [
                      {
                        "x": 1,
                        "y": 1
                      },
                      {
                        "x": 1,
                        "y": 0
                      },
                      {
                        "x": 1.5,
                        "y": 0.9999999999999999
                      }
                    ],
                    "gradientStops": [
                      {
                        "color": {
                          "r": 0.5921568870544434,
                          "g": 0.7137255072593689,
                          "b": 0.5647059082984924,
                          "a": 1
                        },
                        "position": 0
                      },
                      {
                        "color": {
                          "r": 0.3333333432674408,
                          "g": 0.5686274766921997,
                          "b": 0.3450980484485626,
                          "a": 1
                        },
                        "position": 0.5038507580757141
                      },
                      {
                        "color": {
                          "r": 0.13333334028720856,
                          "g": 0.4901960790157318,
                          "b": 0.24313725531101227,
                          "a": 1
                        },
                        "position": 1
                      }
                    ]
                  }
                ],
                "strokeWeight": 1,
                "strokeAlign": "OUTSIDE",
                "effects": [],
                "rectangleCornerRadii": [
                  8,
                  12.5,
                  12.5,
                  12.5
                ]
              }
            ],
            "absoluteBoundingBox": {
              "x": 325,
              "y": 45,
              "width": 102,
              "height": 44
            },
            "constraints": {
              "vertical": "SCALE",
              "horizontal": "SCALE"
            },
            "clipsContent": true,
            "background": [
              {
                "blendMode": "NORMAL",
                "type": "SOLID",
                "color": {
                  "r": 1,
                  "g": 1,
                  "b": 1,
                  "a": 1
                }
              }
            ],
            "backgroundColor": {
              "r": 1,
              "g": 1,
              "b": 1,
              "a": 1
            },
            "layoutGrids": [],
            "effects": []
          }
```